src/listing.rs

```diff
@@ -161,6 +161,14 @@ pub fn directory_listing(
     let current_user: Option<&CurrentUser> = extensions.get::<CurrentUser>();
 
     let conf = req.app_data::<crate::MiniserveConfig>().unwrap();
+    if conf.disable_indexing {
+        return Ok(ServiceResponse::new(
+            req.clone(),
+            HttpResponse::NotFound()
+                .content_type(mime::TEXT_PLAIN_UTF_8)
+                .body("File not found."),
+        ));
+    }
```

Comment:
I think this is probably fine but I'm wondering: Can we just disable indexing entirely in `main.rs`?

Reply:
Technically yes, and if you want to see what that looks like, you can check out [7b5b1a0](https://github.com/svenstaro/miniserve/pull/1329/commits/7b5b1a0e7fec2c0577eb0dc2a93e9fde48228a67). Reiterating what I said [in this comment](https://github.com/svenstaro/miniserve/pull/1329?notification_referrer_id=NT_kwDOABuZp7I5MjQ5MTgzMzUyOjE4MDg4MDc#discussion_r1471362494) I think it would be less complicated to do it like this.